cmd/buildkitd/main.go

```diff
@@ -495,7 +497,7 @@ func getListener(addr string, uid, gid int, tlsConfig *tls.Config) (net.Listener
 	proto := addrSlice[0]
 	listenAddr := addrSlice[1]
 	switch proto {
-	case "unix":
+	case "unix", "npipe":
```

Comment:
I'm not sure we want to expose npipe w/o auth

Reply:
By default, npipe is only accessible to Administrators, as far as I can tell. That's certainly the behaviour I've seen from both buildkitd and containerd: when I run them in an elevated shell, I have to run buildctl/ctr in an elevated shell too. That makes sense, as buildkit is using containerd's listen API.

Another option would be to change the default socket in the Windows config to something other than npipe, but that seems riskier to me.

Reply:
A little bit of reading suggests that default Named Pipes have 'Read' permission for all users, so perhaps a user on the same machine could open the client end of the pipe for read, and see messages from the server. I'm not sure though, the [documentation](https://docs.microsoft.com/en-us/windows/win32/ipc/named-pipe-security-and-access-rights) is a little sparse.

If this is a concern, probably worth flagging through to containerd too.